test/re/frida/MethodTest.java

```diff
@@ -55,14 +59,183 @@ public void replacementPropagatesExceptions() {
+                "}"
+                );
+        assertEquals("ok", script.getNextMessage());
+    }
+    
+    // this one still just producing 
+    // Error: access violation accessing 0xf2b295fe
+    @Test
+    public void nativeReturnGenericVmStack() { 
+        loadScript(
+                "try {" +
+                "  var c = Java.use('dalvik.system.VMStack');" +
+                "  var orig = c.getStackClass2;" +
+                "  c.getStackClass2.implementation = function () {" +
+                "    orig.call(this);" +
+                "  };" +
+                "  var stack = c.getStackClass2();" +
+                "  send('ok');" +
+                "} catch(e) {" + 
+                "  send('nativeReturnGeneric: ' + e);" + 
+                "}"
+                );
+        assertEquals("ok", script.getNextMessage());
+    }
+    
+    // this one still just producing 
+    // Error: access violation accessing 0x2133c66a
+    @Test
+    public void nativeReturnGenericBadgerWrapperAroundJavaLangClass() { 
+        loadScript(
+                "try {" +
+                "  var c = Java.use('re.frida.Badger');" +
+                "  var orig = c.forNameYo;" +
+                "  c.forNameYo.implementation = function () {" +
+                "    orig.call(this);" +
+                "  };" +
+                "  var test = c.forNameYo('re.frida.Badger', false, null);" +
+                "  send('ok');" +
+                "} catch(e) {" + 
+                "  send('nativeReturnGeneric: ' + e);" + 
+                "}"
+                );
+        assertEquals("ok", script.getNextMessage());
+    }
+    
+    // this one was just hanging indefinitely during the test, but in an actual app, it was crashing
+    //! either one of those is bad.
+    /*@Test
+    public void methodInvoke() {
+        loadScript("var c = Java.use('java.lang.reflect.Method');" +
+                "var c2 = Java.use('java.lang.Class');" +
+                "try {" +
+                
+                // hook the original
+                "  var orig = c.invoke;" +
+                "  c.invoke.implementation = function (obj, ...args) { " +
+                "    orig.call(this, obj, args);" +
+                "  };" +
+                
+                // now call it and see what happens
+                "  var cl = c2.forName('re.frida.Badger');" +
+                "  var method = cl.getMethod('returnZero', 'int');" +
+                "  var ret = method.invoke();" +
+                "  send('ok');" +
+                "} catch (e) {" + 
+                "  send('Method.invoke: ' + e);" + 
+                "}"
+                );
+        assertEquals("ok", script.getNextMessage());
+    }*/
+    
+    @Test
+    public void loadWorks() {
+        loadScript("var c = Java.use('java.lang.System');" +
+                "try {" +
+                "  var orig = c.load;" +
+                "  c.load.implementation = function (s) { " +
```

Comment:
This reads out the same value twice making the code hard to read. (See comment below.)